modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/StaleTransactionOperationException.java

```diff
@@ -21,10 +21,10 @@
 import static org.apache.ignite.lang.ErrorGroups.Transactions.TX_STALE_OPERATION_ERR;
 
 import java.util.UUID;
-import org.apache.ignite.tx.TransactionException;
+import org.apache.ignite.internal.tx.TransactionInternalException;
 
 /** Error that occurs when a stale operation of a completed transaction is detected. */
-public class StaleTransactionOperationException extends TransactionException {
+public class StaleTransactionOperationException extends TransactionInternalException {
```

Comment:
Why it's internal? I'd rather move it to public API. Generally I believe that we should go through all our exceptions case by case and define whether they are semantically internal or not.

Reply:
To change public API, we need some discussion. I was planning to do it in IGNITE-20415. I added a TODO for now.

Reply:
Not sure about discussion within context of such changes. Anyway ignite-20415 it is.